packages/react/src/primitives/types/textField.ts

```diff
@@ -32,25 +31,9 @@ export interface TextFieldOptions extends FieldProps, FlexContainerStyleProps {
 
   /**
    * @description
-   * Input field type (only applies when isMultiline = false)
+   * Input field type
    */
   type?: InputProps['type'];
 }
 
-export interface TextInputFieldProps extends TextFieldOptions, InputProps {}
-
-export interface TextAreaFieldMultilineProps
-  extends TextFieldOptions,
-    TextAreaProps {}
-
-export type TextFieldProps<Multiline extends boolean = false> =
```

Comment:
Removing this complexity is what fixes the [change handler typing bug](https://github.com/aws-amplify/amplify-ui/issues/707).